core/dbt/task/compile.py

```diff
@@ -130,9 +134,17 @@ def _runtime_initialize(self):
             )
             sql_node = block_parser.parse_remote(self.args.inline, "inline_query")
             process_node(self.config, self.manifest, sql_node)
+            # keep track of the node added to the manifest
+            self._inline_node_id = sql_node.unique_id
 
         super()._runtime_initialize()
 
+    def after_run(self, adapter, results):
+        # remove inline node from manifest
+        if self._inline_node_id:
+            self.manifest.nodes.pop(self._inline_node_id)
```

Comment:
do we need to reset _inline_node_id as well?

Reply:
Good catch! I had it in some version then lost it due to some branch moving around. 

Also added test!